pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -679,6 +679,72 @@ func addExistsOperator(item model.TagQuery, tagMapType string, not bool) (string
 	return fmt.Sprintf(" AND %s (%s)", notStr, strings.Join(tagOperatorPair, " OR ")), args
 }
 
+func (r *ClickHouseReader) GetEntryPointOperations(ctx context.Context, queryParams *model.GetTopOperationsParams) (*[]model.TopOperationsItem, error) {
+	topOps, err := r.GetTopOperations(ctx, queryParams)
+	if err != nil {
+		return nil, err
+	}
+	if topOps == nil {
+		return nil, errors.New("no top operations found")
+	}
+
+	namedArgs := []interface{}{
+		clickhouse.Named("start", strconv.FormatInt(queryParams.Start.UnixNano(), 10)),
+		clickhouse.Named("end", strconv.FormatInt(queryParams.End.UnixNano(), 10)),
+		clickhouse.Named("service", queryParams.ServiceName),
+		clickhouse.Named("start_bucket", strconv.FormatInt(queryParams.Start.Unix()-1800, 10)),
+		clickhouse.Named("end_bucket", strconv.FormatInt(queryParams.End.Unix(), 10)),
+	}
+
+	// Step 2: Get entry point operation names for the given service
+	entryPointSet := map[string]struct{}{}
+
+	query := fmt.Sprintf(`
+		SELECT 
+			name, 
+			max(timestamp) as ts
+		FROM %s.%s
+		WHERE 
+			timestamp >= @start AND timestamp <= @end
+			AND ts_bucket_start >= @start_bucket AND ts_bucket_start <= @end_bucket
+			AND resource_string_service$$name = @service
+			AND parent_span_id NOT IN (
+				SELECT span_id 
+				FROM %s.%s 
+				WHERE resource_string_service$$name = @service
+			)
+		GROUP BY name
+		ORDER BY ts DESC`,
```

Comment:
This is a rather extremely inefficient way to obtain entry point spans. You already worked span scope where you used some existing information to obtain the entry points spans, why are you implementing this differently?